apiserver/admin_test.go

```diff
@@ -183,7 +183,7 @@ func (s *loginSuite) TestLoginAsDeactivatedUser(c *gc.C) {
 		DisplayName: "Charlie Brown",
 		CreatorUUID: s.AdminUserUUID,
 		Password:    ptr(auth.NewPassword(pass)),
-		Permission:  permission.ControllerForAccess(permission.LoginAccess),
+		Permission:  permission.ControllerForAccess(permission.LoginAccess, s.ControllerUUID),
```

Comment:
You could add a helper method to the APIServerSuite, where the s.ControllerUUID is defined, to replace this logic.

Reply:
True, though I just used a sed script to replace this everywhere. I think its fine to have it explicit, it should hopefully make the tests more readable/debuggable.